crates/rolldown_binding/src/options/plugin/binding_plugin_options.rs

```diff
@@ -84,12 +85,72 @@ pub struct BindingPluginOptions {
   pub write_bundle: Option<MaybeAsyncJsCallback<BindingOutputs, ()>>,
 }
 
-impl Debug for BindingPluginOptions {
+impl<const WEAK: bool> Debug for BindingPluginOptions<WEAK> {
   fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
     f.debug_struct("BindingPluginOptions").field("name", &self.name).finish_non_exhaustive()
   }
 }
 
+impl FromNapiValue for BindingPluginOptions<true> {
```

Comment:
napi_derive doesn't allow us to write it like:
```rust
#[napi_derive::napi(object, object_to_js = false)]
#[derive(Deserialize, Default)]
#[serde(rename_all = "camelCase")]
pub struct BindingPluginOptions<const WEAK: bool = false> {
  #[serde(skip_deserializing)]
  #[napi(ts_type = "(ctx: BindingPluginContext) => MaybePromise<VoidNullable>")]
  pub build_start: Option<MaybeAsyncJsCallback<BindingPluginContext, (), WEAK>>,
}
```
Is there a way to make this work in napi_derive?

Reply:
@Brooooooklyn Does threadsafe_function `unref` by default?

Reply:
Yes if `WEAK` is `true`